README.md

```diff
@@ -3,30 +3,209 @@
+and Private Key (as `key`) for your account.
+Also you will need to download the Chain CA using the button
+to use it as `ca`.
+
+### Sender Options
+
+These options are passed to the `devo.sender()` constructor.
+They will determine where and how to send the events.
+
+#### `host`
+
+Host to send the events.
+Optional, default value `localhost`.
+Can be looked up in Devo as
+[relay
+address](https://docs.devo.com/confluence/docs/system-configuration/relays),
+as (host:port).
+
+#### `port`
+
+Port to send the events.
+Mandatory.
+Can be looked up in Devo as
+[relay
+address](https://docs.devo.com/confluence/docs/system-configuration/relays),
+as (host:port).
+
+#### `cert`
+
+Certificate for sending events securely.
+Optional; if not present will send events as cleartext (not recommended).
+You need to pass the whole certificate, not a path.
+
+#### `key`
+
+Private key for sending events securely.
+Optional; must be present if `cert` is used.
+You need to pass the whole key, not a path.
+
+#### `ca`
 
-Import in your code and create the client:
+Certificate of the authority that created `cert` for sending events securely.
+Optional; needed only if the `cert` is not signed by a recognized CA.
+You need to pass the whole certificate, not a path.
+
+#### `objectMode`
+
+If the sender is going to be used as a stream and you want to
+send objects, set to `true`. Objects will be converted to JSON.
+Optional, default `false`.
+See [send to stream](#send-to-stream) for details.
+
+#### `priority`
+
+Priority value as specified by
+[RFC 5424](https://tools.ietf.org/html/rfc5424#section-6.2.1).
+Optional, default value `13`.
+
+#### `tag`
+
+Tag to use for the events.
+The tag will determine the table to store the events on Devo.
+Optional, default value `my.app`.
+
+#### `localhost`
+
+Name of the host that generates the events.
+Optional, default value `localhost.localdomain`.
+
+### Putting It All Together
+
+To create the sender you will need to pass all the parameters required.
+Example:
+
+```js
+const sender = devo.sender({
+  host: 'app.logtrust.com',
+  port: 443,
+  cert: fs.readFileSync('path/to/cert'),
+  key: fs.readFileSync('path/to/key'),
+})
+```
+
+### Send to Stream
+
+The `sender` returned can be used as a writeable Node.js stream:
+it can be written to, piped to and so on.
+The stream also implements backpressure:
+if too much data is sent at a time it will throttle the writer back.
+
+For example, to send a file to Devo line by line:
+
+```js
+const devo = require('@devo/nodejs-sdk')
+const fs = require('fs')
+
+const sender = devo.sender(options)
+const rs = fs.createReadStream('/path/to/file')
+rs.pipe(sender)
+```
+
+It can also be used to send objects, which will be converted to JSON
+before sending.
+In this case the option `objectMode` must be set to `true`:
+
+```js
+const sender = devo.sender({
+  objectMode: true,
+  ...
+})
+sender.write({key: value})
+sender.end()
+```
```

Comment:
still confusing: `objectMode` is set to `true`, but the sender is not used as a stream.

Reply:
??? [`stream.write()`](https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback) is a stream method, just as `stream.pipe()`.